core/src/main/java/org/apache/carbondata/core/indexstore/AbstractMemoryDMStore.java

```diff
@@ -33,6 +33,8 @@
 
   protected final String taskId = ThreadLocalTaskInfo.getCarbonTaskInfo().getTaskId();
 
+  private long totalRowCount;
```

Comment:
Please keep in unsafe, I don't think get from unsafe now causes any delay.

Reply:
done